src/HealthGPS/analysis_module.cpp

```diff
@@ -304,6 +305,47 @@ DALYsIndicator AnalysisModule::calculate_dalys(Population &population, unsigned
                           .disability_adjusted_life_years = yll + yld};
 }
 
+void AnalysisModule::calculate_population_statistics(RuntimeContext &context) const {
+    auto num_factors_to_calculate =
+        context.mapping().entries().size() - factors_to_calculate_.size();
+
+    for (const auto &entity : context.population()) {
+        // Get factors_to_calculate_ values for this entity
+        std::vector<double> entity_factors;
+        for (const auto &factor : factors_to_calculate_) {
+            entity_factors.push_back(entity.get_risk_factor_value(factor));
+        }
+        // Get the bin index for each factor
+        std::vector<int> bin_indices;
+        for (size_t i = 0; i < factors_to_calculate_.size(); i++) {
+            auto factor_value = entity_factors[i];
```

Comment:
I think this vector can be removed entirely:
```suggestion
        // Get the bin index for each factor
        std::vector<int> bin_indices;
        for (size_t i = 0; i < factors_to_calculate_.size(); i++) {
            double factor_value = entity.get_risk_factor_value(factors_to_calculate_[i]);
```